core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -1,56 +1,290 @@
 import org.stellar.anchor.config.Sep31Config;
+import org.stellar.anchor.dto.sep12.Sep12GetCustomerRequest;
+import org.stellar.anchor.dto.sep12.Sep12GetCustomerResponse;
 import org.stellar.anchor.dto.sep31.*;
+import org.stellar.anchor.dto.sep31.Sep31GetTransactionResponse.TransactionResponse;
+import org.stellar.anchor.exception.AnchorException;
+import org.stellar.anchor.exception.BadRequestException;
+import org.stellar.anchor.exception.NotFoundException;
+import org.stellar.anchor.exception.SepException;
 import org.stellar.anchor.integration.customer.CustomerIntegration;
-import org.stellar.anchor.integration.rate.RateIntegration;
+import org.stellar.anchor.integration.fee.FeeIntegration;
+import org.stellar.anchor.model.Sep31Transaction;
+import org.stellar.anchor.model.Sep31TransactionBuilder;
+import org.stellar.anchor.model.Sep38Quote;
+import org.stellar.anchor.model.TransactionStatus;
+import org.stellar.anchor.sep10.JwtToken;
+import org.stellar.anchor.sep38.Sep38QuoteStore;
+import org.stellar.platform.apis.callbacks.requests.GetFeeRequest;
+import org.stellar.platform.apis.callbacks.responses.GetFeeResponse;
 
 public class Sep31Service {
+  private final AppConfig appConfig;
   private final Sep31Config sep31Config;
+  private final Sep31TransactionStore sep31TransactionStore;
+  private final Sep38QuoteStore sep38QuoteStore;
   private final AssetService assetService;
-  private final RateIntegration rateIntegration;
+  private FeeIntegration feeIntegration;
   private final CustomerIntegration customerIntegration;
   private Sep31InfoResponse infoResponse;
 
   @SneakyThrows
   public Sep31Service(
+      AppConfig appConfig,
       Sep31Config sep31Config,
+      Sep31TransactionStore sep31TransactionStore,
+      Sep38QuoteStore sep38QuoteStore,
       AssetService assetService,
-      RateIntegration rateIntegration,
+      FeeIntegration feeIntegration,
       CustomerIntegration customerIntegration) {
+    this.appConfig = appConfig;
     this.sep31Config = sep31Config;
+    this.sep31TransactionStore = sep31TransactionStore;
+    this.sep38QuoteStore = sep38QuoteStore;
     this.assetService = assetService;
-    this.rateIntegration = rateIntegration;
+    this.feeIntegration = feeIntegration;
     this.customerIntegration = customerIntegration;
-
-    infoResponse = createFromAsset(assetService.listAllAssets());
+    this.infoResponse = createFromAsset(assetService.listAllAssets());
   }
 
   public Sep31InfoResponse getInfo() {
     return infoResponse;
   }
 
-  public Sep31PostTransactionResponse postTransaction(Sep31PostTransactionRequest request) {
-    return null;
+  public Sep31PostTransactionResponse postTransaction(
+      JwtToken jwtToken, Sep31PostTransactionRequest request) throws AnchorException {
+
+    validateAmount("", request.getAmount());
+    validateLanguage(appConfig, request.getLang());
+    validateRequiredFields(request.getAssetCode(), request.getFields().getTransaction());
```

Comment:
That is right. The logic now is in SEP implementation to check required fields that we return from `/info`. We should probably define an API for transaction validation. And in that API, we also remove the logic of transaction validation in the `/fee` API.

I currently cannot test the `PATCH /transactions` endpoint yet. Will do more when we have payment observer and platform API in place.

Reply:
Why do we need to test the `PATCH /transactions` endpoint? I suggesting we remove it, since anchors using the Platform cannot access per-transaction fields anyway.